packages/core/src/transforms/resize.ts

```diff
@@ -24,28 +45,46 @@ export const resize: TransformFactory<ResizeOptions> = (config, ctx) => {
         : config.h
             ? parseInt(config.h)
             : undefined
+    const aspect = width && height
+        ? width / height
+        : parseAspect(config.aspect)
 
-    if (!width && !height) return
+    if (!width && !height && !aspect) return
 
     return function resizeTransform(image) {
+        let finalWidth = width
+        let finalHeight = height
+
+        const w = getMetadata(image, 'width')
+        const h = getMetadata(image, 'height')
 
-        if (!height) {
-            const w = getMetadata(image, 'width')
-            const h = getMetadata(image, 'height')
-            setMetadata(image, 'height', Math.round((width! / w) * h))
-            setMetadata(image, 'width', width)
+        const metaAspect = w / h
+
+        if (width && height) {
+            /* both dimensions were provided, aspect is ignored and no calculations are needed */
         }
+        else if (!height && !width) {
+            /* only aspect was given, need to calculate which dimension to crop */
+            const useWidth = aspect! > metaAspect
 
-        if (!width) {
-            const w = getMetadata(image, 'width')
-            const h = getMetadata(image, 'height')
-            setMetadata(image, 'width', Math.round((height! / h) * w))
-            setMetadata(image, 'height', height)
+            finalHeight = useWidth ? Math.round(w / aspect!) : h
+            finalWidth = useWidth ? w : Math.round(h / aspect!)
+        } else if (!height) {
+            /* only width was provided, need to calculate height */
+            finalHeight = width! / (aspect || metaAspect)
+            finalWidth = width
+        } else {
+            /* only height was provided, need to calculate width */
+            finalHeight = height
+            finalWidth = height * (aspect || metaAspect)
         }
 
+        setMetadata(image, 'height', finalHeight)
+        setMetadata(image, 'width', finalWidth)
+
```

Comment:
Maybe you could add an aspect-ratio metadata property just for good measure, I could image that being useful to people importing the image, sure I'd be a floating point number then, but people could make use of that still.
```suggestion
setMetadata(image,'aspect',aspect)
```

Reply:
Yep that's a great idea, done!